plugins/inputs/cnosdb/kv_service.go

```diff
@@ -32,97 +31,96 @@ func NewTSKVService(acc telegraf.Accumulator) service.TSKVServiceServer {
-			if !points.Tables(&table, tblI) {
+		for {
+			r, err := recordReader.Read()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				log.Printf("failed to read record data: %v", err)
+				s.accumulator.AddError(fmt.Errorf("failed to read record data: %w", err))
 				continue
 			}
-			table.Schema(&schema)
-			tab := string(table.TabBytes())
-			for ptI := 0; ptI < table.PointsLength(); ptI++ {
-				if !table.Points(&point, ptI) {
-					continue
+
+			numRows := r.NumRows()
+			numCols := r.NumCols()
+			colArrMap := make([]interface{}, len(tableschema.Columns))
+
+			for j, col := range r.Columns() {
+				switch columnIndexToType[j] {
+				case "TAG_STRING":
+					colArrMap[j] = array.NewStringData(col.Data())
+				case "FIELD_STRING":
+					colArrMap[j] = array.NewStringData(col.Data())
+				case "FIELD_BIGINT":
+					colArrMap[j] = array.NewInt64Data(col.Data())
+				case "FIELD_BIGINT UNSIGNED":
+					colArrMap[j] = array.NewUint64Data(col.Data())
+				case "FIELD_DOUBLE":
+					colArrMap[j] = array.NewFloat64Data(col.Data())
+				case "FIELD_BOOLEAN":
+					colArrMap[j] = array.NewBooleanData(col.Data())
 				}
+			}
 
+			for i := 0; i < int(numRows); i++ {
 				tags := make(map[string]string)
 				fields := make(map[string]interface{})
 
-				tagsBitSet := BitSet{
-					buf: point.TagsNullbitBytes(),
-					len: schema.TagNameLength(),
-				}
-				if point.TagsLength() > 0 {
-					for tagI := 0; tagI < point.TagsLength(); tagI++ {
-						if !tagsBitSet.get(tagI) {
-							continue
-						}
-						if !point.Tags(&tag, tagI) {
-							continue
-						}
-						tags[string(schema.TagName(tagI))] = string(tag.ValueBytes())
-					}
-				}
-				fieldsBitSet := BitSet{
-					buf: point.FieldsNullbitBytes(),
-					len: schema.FieldNameLength(),
-				}
-				if point.FieldsLength() > 0 {
-					for fieldI := 0; fieldI < point.FieldsLength(); fieldI++ {
-						if !fieldsBitSet.get(fieldI) {
-							continue
-						}
-						if !point.Fields(&field, fieldI) {
-							continue
-						}
-						switch schema.FieldType(fieldI) {
-						case models.FieldTypeInteger:
-							v := binary.BigEndian.Uint64(field.ValueBytes())
-							fields[string(schema.FieldName(fieldI))] = int64(v)
-						case models.FieldTypeUnsigned:
-							v := binary.BigEndian.Uint64(field.ValueBytes())
-							fields[string(schema.FieldName(fieldI))] = v
-						case models.FieldTypeFloat:
-							tmp := binary.BigEndian.Uint64(field.ValueBytes())
-							v := math.Float64frombits(tmp)
-							fields[string(schema.FieldName(fieldI))] = v
-						case models.FieldTypeBoolean:
-							v := field.ValueBytes()
-							fields[string(schema.FieldName(fieldI))] = v[0] == byte(1)
-						case models.FieldTypeString:
-							v := string(field.ValueBytes())
-							fields[string(schema.FieldName(fieldI))] = v
-						default:
-							// Do nothing ?
-						}
+				for j := 0; j < int(numCols); j++ {
+					switch columnIndexToType[uint64(j)] {
```

Comment:
Missing variants of time column, such as  `TIME_TIMESTAMP(NANOSECOND)`

Reply:
Done.